.github/workflows/ci_test.yml

```diff
@@ -0,0 +1,133 @@
+name: e_e CI targets
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  release:
+    types:
+      - published
+
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install non-python dependencies on mac
+      if: runner.os == 'macOS'
+      run: |
+        brew install gcc
+        brew install automake suite-sparse
+        curl -sSL https://raw.githubusercontent.com/vallis/libstempo/master/install_tempo2.sh | sh
+    - name: Install non-python dependencies on linux
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get install libsuitesparse-dev
+        curl -sSL https://raw.githubusercontent.com/vallis/libstempo/master/install_tempo2.sh | sh
+    - name: Install dependencies and package
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install flake8 pytest black pytest-cov Cython numpy scipy
+        python -m pip install -e .
+    - name: Display Python, pip, setuptools, and all installed versions
+      run: |
+        python -c "import sys; print(f'Python {sys.version}')"
+        python -c "import pip; print(f'pip {pip.__version__}')"
+        python -c "import setuptools; print(f'setuptools {setuptools.__version__}')"
+        python -m pip freeze
+    #- name: Run lint
+    #  run: make lint
+    - name: Test with pytest
+      run: make test
+    - name: Codecov
+      uses: codecov/codecov-action@v1
+      #with:
+      #  fail_ci_if_error: true
+
+
+  build:
+    needs: [tests]
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release'
```

Comment:
Will do, though right now the tests aren't passing in my fork because of an issue with `gcc`. I originally was using the command you had in the `enterprise` script:
```
brew unlink gcc && brew link gcc
```
and I also tried
```
brew install gcc
```
but Actions still can't seem to [find](https://github.com/Hazboun6/enterprise_extensions/actions/runs/776711208) `gcc` and is trying to use clang. 

These `Cython`/compiler dependencies are all new and are for the outlier code that @svigeland's student, @gabefreedman just contributed. Everything works in linux of course :-)

Reply:
yea I don't know why thats happening. I saw that before but the unlink and linking worked. Maybe run `brew reinstall gcc` first?